caste.go

```diff
@@ -612,14 +696,29 @@ func ToUint8E(i interface{}) (uint8, error) {
 		}
 		return 0, fmt.Errorf("unable to cast %#v to uint8: %s", i, err)
 	case int:
+		if s < 0 {
+			return 0, fmt.Errorf("unable to cast negative value to uint8")
```

Comment:
I was trying to avoid allocating the error unless I needed it, but I can refactor if you want.  Let me know what you want to do.